### ISSUE TITLE
Tab is not hitting tree component

### ISSUE THREAD

**Issue by debslab** [500tech/angular-tree-component#199]
I have tree component in my page. When I press tab, I don't see it's hitting tree component. I have to use mouse to select one of the node, and then only I can do keyboard navigation which seems to be usability issue to me. Please let me know if I am missing something.

**Comment by debslab** [comment]
May I get some update here please, at least a pointer to how I can achieve it by either modifying this component or from outside.

**Comment by adamkleingit** [comment]
Hi,
I can't reproduce, in the example it works for me.
Can you share your code?

**Comment by debslab** [comment]
Please check following plunkr. https://plnkr.co/edit/Zm98g3kT6yU7XWcUwTvX?p=preview.

If you hit tab, button will be highlighted and tree component is never touched

**Comment by adamkleingit** [comment]
Hi,
I hope to have time to look at it during the weekend

-- 
- Adam Klein <http://il.linkedin.com/in/adamklein500>
- CTO @ 500Tech.com <http://500tech.com/>
- AngularJS, ReactJS, NodeJS
- Development, consulting, training and public speaking
- (+972) 52 747 5633

**Comment by AngularTx** [comment]
Any updates here?? I too have the same issue .

**Comment by AngularTx** [comment]
Any updates here ?

**Comment by AngularTx** [comment]
hi ,

any update here? even after setting the focus on the first node , key navigation doesn't work. we need to click the node and then the traversing happens. 

And also the tab doesn't on tree .

**Comment by adamkleingit** [comment]
Hey @AngularTx,
Did you try to debug it?

**Comment by AngularTx** [comment]
nope. during testing I found that .

Did you get time to take a look at the issue ?

**Comment by adamkleingit** [comment]
Got a big issue backlog. It will be very helpful if you try to debug it -
thanks!

-- 
- Adam Klein <http://il.linkedin.com/in/adamklein500>
- CTO @ 500Tech.com <http://500tech.com/>
- AngularJS, ReactJS, NodeJS
- Development, consulting, training and public speaking
- (+972) 52 747 5633

**Comment by Suncrusher** [comment]
I did have the same problem. My solution looks like this:

I add an eventlistener to the "enter" key. Only then the user can use left and right arrow to navigate. When "enter" is pressed I mark the node I want with this function:

```
  markNode(idOfNode) {
    this.tree.setFocus(true);
    const node = this.tree.getNodeById(idOfNode);
    this.tree.setActiveNode(node, true, false);
    const activeNode = this.tree.getActiveNode();
    let myElement = document.querySelector('#l-' + activeNode.data.id) as HTMLElement;
    this.renderer.invokeElementMethod(myElement, 'focus');
  }
```

The key to this is the 
`this.tree.setActiveNode(node, true, false);`

Then the arrow keys work. In my template I do have a invisible checkbox and a label. I do want the label to receive the focus.

Hope this helps.

**Comment by sadiksha** [comment]
Hi,
I have the same issue now as @debslab  mentioned. However, @Suncrusher's idea does not solve the original issue of tree not being selected in the beginning. I also have the same issue that first I have to click on the tree node then I can use the keyboards for navigation, however, I also tried this: 
` <tree-root [nodes]="nodes" [focused]="true" [options]="options"></tree-root>` which also doesn't seem to work. Am I missing something here?
